src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -247,6 +247,10 @@ private static Assembly ResolveForKnownRequestor(string requestedAssemblyName, A
 
                     string directoryPath = Path.GetDirectoryName(requestingAssemblyFullPath);
                     string assemblyFullPath = Path.Combine(directoryPath, assemblyIdentity.Name + ".dll");
+                    if (!File.Exists(assemblyFullPath))
+                    {
+                        return null;
+                    }
 
```

Comment:
For 2. it seems CodeBase is non-null for our CodeAnalysis analyzers assembly in both IDE and command line scenarios. 
For command line it points to "file:///c:/roslyn/open/binaries/debug/Microsoft.CodeAnalysis.Desktop.dll", which is the location from where I am running csc. For IDE it points to "file:///C:/USERS/MAVASANI/APPDATA/LOCAL/MICROSOFT/VISUALSTUDIO/14.0ROSLYNDEV/EXTENSIONS/MICROSOFT/ROSLYN%20TELEMETRY/42.42.42.42/Microsoft.CodeAnalysis.Desktop.dll"

I'd like to debug further into this to find out what is causing the different code paths for resource resolution, but CLR assemblies are all optimized making debugging/setting breakpoints almost impossible. I think it is best if we talk to experts on CLR team to understand the root cause of both these issues, rather then trying to debug through it. I'll follow up offline on this.